prototypes/process-explorer/oss-frontend/src/app/components/processes/processes.component.ts

```diff
@@ -21,28 +23,36 @@ import {LiveAnnouncer} from '@angular/cdk/a11y';
     ]),
   ],
 })
-export class ProcessesComponent implements AfterViewInit {
+export class ProcessesComponent {
   expandedElement: any;
-  processesData: Array<ProcessTable>;
-  dataSource = new MatTableDataSource<ProcessTable>()
+  processesData: Array<ProcessTable>; 
+  dataSource = new MatTableDataSource<Process.AsObject>();
+  allProcesses: Array<Process.AsObject> = [];
 
-  displayedColumns = [{ key: 'ProcessName', header: 'Process Name' }, { key: 'PID', header: 'PID' }, { key: 'ProcessStatus', header: 'Process Status' }, { key: 'StartTime', header: 'Start Time' }, { key: 'ProcessorUsage', header: 'Processor Usage' }, { key: 'PhysicalMemoryUsageBit', header: 'Physical Mem Usage' }, { key: 'PriorityLevel', header: 'Priority Level' }, { key: 'VirtualMemorySize', header: 'Virtual Memory Size' }];
+  displayedColumns = [{ key: 'processname', header: 'Process Name' }, { key: 'processid', header: 'PID' }, { key: 'processstatus', header: 'Process Status' }, { key: 'starttime', header: 'Start Time' }, { key: 'processorusage', header: 'Processor Usage' }, { key: 'physicalmemoryusagebit', header: 'Physical Mem Usage' }, { key: 'processpriorityclass', header: 'Priority Level' }, { key: 'virtualmemorysize', header: 'Virtual Memory Size' }];
   displayedColumnsKeys: string[];
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
 
   constructor(private processService: ProcessesService, private liveAnnouncer: LiveAnnouncer) {
-    this.processService.getProcesses('Processes').subscribe(process => this.processesData = process);
-    this.displayedColumnsKeys = this.displayedColumns.map(column => column.key);
-    this.dataSource = new MatTableDataSource<ProcessTable>(this.processesData);
-  }
+    this.processService.getProcessesData().on("data", req => {
```

Comment:
Right now it's fine to do this but this old ProcessExplorer backend currently sends message with actiontype. ActionType can be considered basically like topics eg.: UpdateProcessAction, UpdateProcessStatusAction, etc.. which determines the message. It would be -I think nice to have it seperated that which service watch or does something based on the topic. 
This could be an improvement later, also the topics should be revisted on the backend side as the functionality as somewhere it is broken - especially if we add the Cors within the ProcessExplorerBuilderExtensions.class, also it has some weird things like ModifySubsystemAction (this is not possible with the use of the new ModuleLoader eiter it wasn't with the old one..) and also the structure of the backend needs to be revisited/rewritten. Also we shouldn't expose the service itself but that is the part of this improvement. :)